Lab_Management/PowerShell/Quota/Get-LabServicesCapacityQuotas.ps1

```diff
@@ -0,0 +1,104 @@
+}
+
+# Let's figure out all regions where Lab Services is used:
+$restUri = "https://management.azure.com/subscriptions/$subscriptionId/providers/microsoft.labservices?api-version=2014-04-01-preview"
+$response = Invoke-RestMethod -Uri $restUri -Method Get -Headers $authHeader
+
+# List of all regions available for lab services
+$availableRegionsForLabs = ($response.resourceTypes | Where-Object {$_.resourceType -ieq "labplans"}).Locations
+
+# Save a table of region short names to display names
+$locations = Get-AzLocation
+
+$quotas = $availableRegionsForLabs | ForEach-Object {
+    $regionDisplayName = $_
+    # Get the short code for the region:
+    $region = ($locations | Where-Object {$_.DisplayName -ieq $regionDisplayName} | Select -First 1).Location
+    
+    $authHeader = @{
+        'Content-Type'='application/json'
+        'Authorization'='Bearer ' + $((Get-AzAccessToken).Token)
+    }
+
+    $restUri = "https://management.azure.com/subscriptions/$subscriptionId/providers/Microsoft.LabServices/locations/$region/usages?api-version=2021-11-15-preview"
+    try {
+        $response = Invoke-RestMethod -Uri $restUri -Method Get -Headers $authHeader
+
+        $response.value | Where-Object {$_.limit -gt 0} |
+                Select-Object -Property `
+                    @{Name = 'Region'; Expression = {$region}},
+                    @{Name = 'RegionName'; Expression = {$regionDisplayName}},
+                    @{Name = 'Size'; Expression = {$_.name.value}},
+                    @{Name = "LabSize"; Expression = {$labsSizes[$_.name.value].Keys -Join ", "}},
+                    @{Name = 'UsedCores'; Expression = {$_.currentValue}},
+                    @{Name = 'TotalCores'; Expression = {$_.limit}},
+                    @{Name = 'PercentUsed'; Expression = {[Math]::Round($_.currentValue / $_.limit * 100)}},
+                    @{Name = 'AvailableVMs'; Expression = {
+                            $availableCores = $_.limit - $_.currentValue
+                            ($labsSizes[$_.name.value].GetEnumerator() | ForEach-Object {
+                                "$([Math]::Floor($availableCores / $_.Value)) $($_.Name)"
+                            }) -join ', ' }}
+                            
+    }
+    catch [System.Net.WebException] {
+        # There are a few regions where labs isn't deployed, so we get errors for those that we can ignore
+    }
+}
+
+Write-Host "Quotas per Size:" -ForegroundColor Green
+$quotas | Group-Object -Property LabSize | 
+    Where-Object {$_.Name -and $_.Name -ine "labPlans" -and $_.Name -ine "labs"} |  # There are some sizes we have quotas for like Esv2 that aren't mapped
+    ForEach-Object {
+        Write-Host "Lab Sizes:  $($_.Name)" -ForegroundColor Cyan
+        $_.Group | 
+            Select-Object -Property RegionName, UsedCores, TotalCores, PercentUsed, AvailableVMs | 
+            Format-Table -AutoSize
+    }
+
+if ($OutputCSV) {
```

Comment:
I wonder if it's necessary to add this to the script.  Would it be better to return the quotas table and let the user format the output as they need?

Reply:
Sure, we can change this.  I can update the quota script to print out the details via write-host and then return the data.  I'll get this checked in soon.